.github/workflows/feature-serverless-framework-deployment.yml

```diff
@@ -96,13 +105,14 @@ jobs:
         working-directory: ${{env.working-directory}}
         run: ./main --o latency-samples --c ../experiments/tests/aws/hellopy.json --s true
 
-
   unit_tests:
     name: Unit tests
-    needs: [ build_client ]
+    needs: [ build_client, e2e_aws ]
     runs-on: ubuntu-22.04
     env:
       working-directory: ./src
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_KEY }}
```

Comment:
Not sure if this is necessary any more, if it works without this we can remove it